spec/features/product_description_page_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+
+RSpec.describe "Product Show page", type: :feature do
+  context "when in the product show page" do
+    before(:each) do
+      2.times do
+        product = create(:product)
+        product2 = create(:product, subcategory: product.subcategory)
+        create(:product_image_link, product: product)
+        create(:product_image_link, product: product2)
+        2.times { create(:review, product: product) }
+      end
+    end
+
+    it "contains the created product" do
+      product = Product.first
+
+      visit product_path(product)
+      expect(page).to have_content product.name
+      expect(page).to have_content product.brand
+      expect(page).to have_content product.category.name
+      expect(page).to have_content product.subcategory.name
+
+      expect(page).to have_selector(:link_or_button, "ADD TO CART")
+      expect(page).to have_selector(
+        "img[src='#{product.product_image_links.first.link_name}']"
+      )
+    end
+
+    it "displays related products in the product page" do
+      product = Product.first
+      related_product = product.category.products.last
+
+      visit product_path(product)
+
+      within(:css, "div.related-products") do
+        expect(page).to have_content related_product.name
+        expect(page).to have_content related_product.price
+        expect(page).to have_selector("img")
+      end
+    end
+
+    it "displays the reviews for the current product" do
+      product = Product.first
```

Comment:
Instead of having `product = Product.first` in each of the examples, you could include it in your `before(:each)`, just outside of the `2.times do` loop.